### DISCUSSION TITLE
Pro board status?

### DISCUSSION THREAD

**Discussion by frdfsnlght** [phil-barrett/grblHAL-teensy-4.x#101] [Q&A]
Phil, we spoke years ago about a "pro" version of the your board when I purchased your current BOB. Back then it seemed you were close to completing the thing but I haven't seen any updates since 2022 (discussion #43).

I'd really like to get my hands on the pro version we discussed way when then. I have a new plasma machine I plan on building in the coming months. I'd like to use a pro board with it's vertical pluggable connectors as space is an issue for me, as well your prototype ADC. Now that your pico board is done, are there any plans for the pro board or should I look elsewhere (the [FlexiHAL](https://github.com/Expatria-Technologies/Flexi-HAL) board looks nice)?

**Comment by frdfsnlght** [comment]
Haven't heard anything yet. May I need to ping @phil-barrett so he'll notice the question? Or is he just unavailable for the while?

**Comment by phil-barrett** [comment]
Sorry for taking so long to answer the question.  

I decided not to pursue the pro board due to several factors.  I was very unhappy with the power supply set up - too many inputs and required use of a DC-DC converter was very messy (sourcing and qualifying modules plus I was not convinced of the reliability).  Also, the pluggable screw terminals are pricey which, with 32 of them, made the board fairly expensive compared to other products out there.  I tried to find a way to make a board that could use either the original screw terminals or the pluggable ones but it just didn't work - the shell sizes are different. 

I am working on an update to the current Teensy based board.  It will have a similar power supply structure to the Pico board (12V) and support spindle sync.  And, with the same daughter board (I2C and UART) format. I am still fretting over a couple of other features. One of them is how to make switching between USB and on-board power.  The Pico is super well designed that way - allows use of a single PMOSFET to transparently switch power on the fly.  The Teensy doesn't.  I may just require the user cut the USB voltage trace on the Teensy. Usability implications...

**Comment by frdfsnlght** [comment]
Sad to hear about the pro-board. I really need the pluggable terminals and 12V sensor inputs.

Sounds like I might have to go with the FlexiHAL board. The only issue there is that it's I2C interface is different than yours, but that's not surprising. I'd like to use your THC board, but I'll probably have to design my own (based on your design) to match the pinout on the FlexiHAL.

Thanks for the answers anyway. I'm sure I'm not the only one who wanted to get an update.

**Comment by phil-barrett** [comment]
The new version of the teensy board will support 12V sensor inputs. I like pluggable a lot, too, but the cost was just too high.
